core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2792,9 +2793,9 @@ wasm_runtime_set_wasi_args_ex(WASMModuleCommon *module, const char *dir_list[],
     wasi_args->env_count = env_count;
     wasi_args->argv = argv;
     wasi_args->argc = (uint32)argc;
-    wasi_args->stdio[0] = stdinfd;
-    wasi_args->stdio[1] = stdoutfd;
-    wasi_args->stdio[2] = stderrfd;
+    wasi_args->stdio[0] = (os_raw_handle)stdinfd;
+    wasi_args->stdio[1] = (os_raw_handle)stdoutfd;
+    wasi_args->stdio[2] = (os_raw_handle)stderrfd;
```

Comment:
actually reverting this - there isn't a compilation warning and the intermediate cast to `intptr_t` is not necessarily correct since the size of a platform handle isn't always going to dictated by the pointer size on the machine.